### DISCUSSION TITLE
Multi select "group" feature should be more flexible

### DISCUSSION THREAD

**Discussion by Picoquio** [primefaces/community#1099] [PrimeNG] [answered]
Hi there! First of all thank you for all the work you guys put into this amazing library.

Now, with the issue: if one wants to use the "group" feature of the multi select component, the _options_ object that feeds such component must have certain properties named in a specific way: as far as I figured: 

- _Label_
- _Items_

From the documentation, the example that works:
` this.groupedCities = [
            {
                label: 'Germany',
                value: 'de',
                items: [
                    { label: 'Berlin', value: 'Berlin' },
                    { label: 'Frankfurt', value: 'Frankfurt' },
                    { label: 'Hamburg', value: 'Hamburg' },
                    { label: 'Munich', value: 'Munich' }
                ]
            },`

Now, if you change the name of _label_ and _item_ properties it just stops working
` this.groupedCities = [
            {
                label: 'Germany',
                value: 'de',
                cityList: [
                    { name: 'Berlin', value: 'Berlin' },
                    { name: 'Frankfurt', value: 'Frankfurt' },
                    { name: 'Hamburg', value: 'Hamburg' },
                    { name: 'Munich', value: 'Munich' }
                ]
            },`
            
It would be nice if one could just specify in the _p-multiSelect_ tag which property represents _item_, _label_ and so on. As how it is now, you need to modify whatever data comes from the backend in order to fit into this particular way of passing data to the component. Also, it would be helpful to make this situation clear in the documentation, it took me some time to figure out what was going on

**Comment by Uzeyir9695** [comment, marked as answer]
Hi, it's pretty simple to suit your option or groupOption keys.
If you look in the components props you will see these props: ` optionLabel="label" optionGroupLabel="label" ` 
in your case you just need to set `optionLabel="label"` to `optionLabel="name"`
If you want to change groupOption key name just modify from: ` this.groupedCities = [ { label: 'Germany' ...` to `this.groupedCities = [ { name: 'Germany'` and in the MultiSelect component props change `optionGroupLabel="label"` to `optionGroupLabel="name"`

**Reply by Picoquio** [reply]
Thank you for your response @Uzeyir9695! It worked with a catch: I also had to set `optionGroupChildren="cityList"`. My bad, I could have investigated the props intellisense a little bit more.

**Reply by Uzeyir9695** [reply]
Great, I am happy that it helped you. ))
